pages/snake/index.js

```diff
@@ -0,0 +1,361 @@
+  position: absolute;
+  bottom: 25vh;
+`;
+
+const StyledHowToPlay = styled.div`
+  display: grid;
+  place-content: center;
+  gap: 20px;
+  background: var(--secondary-color);
+  font-size: 0.8rem;
+  line-height: 2;
+  width: 100%;
+  padding: 20px;
+  border-radius: 20px;
+  position: absolute;
+  top: -20px;
+  z-index: 2;
+
+  @media screen and (min-width: 900px) {
+    display: block;
+    width: 120px;
+    right: -140px;
+    top: 20px;
+    font-size: 0.6rem;
+    background: none;
+    padding: 0;
+    line-height: 1.5;
+  }
+
+  @media screen and (min-width: 1200px) {
+    width: 200px;
+    font-size: 0.8rem;
+    right: -230px;
+  }
+`;
+
+export default function SnakeGame({ currentPet, onUpdatePetIndicator }) {
+  const [gameOn, setGameOn] = useState(true);
+  const [playerPosition, setPlayerPosition] = useState({ x: 140, y: 140 });
+  const [children, setChildren] = useState([]);
+  const [foodPosition, setFoodPosition] = useState({});
+  const [direction, setDirection] = useState("");
+  const [prevDirection, setPrevDirection] = useState("");
+  const [score, setScore] = useState(0);
+  const [highscore, setHighscore] = useState(0);
+  const [instructions, setInstructions] = useState(false);
+
+  useEffect(() => {
+    setFoodPosition(generateNewFoodPosition());
+  }, []);
+
+  useEffect(() => {
+    if (!gameOn) {
+      const newHappinessValue = Math.min(
+        currentPet.status.happiness + score,
+        100
+      );
+      onUpdatePetIndicator(newHappinessValue);
+    }
+  }, [gameOn]);
+
+  useEffect(() => {
+    function movePlayer() {
+      if (!gameOn) return;
+      setPlayerPosition((prevPosition) => {
+        let newPosition = { ...prevPosition };
+
+        if (direction === "ArrowUp") {
+          newPosition.y -= 20;
+        } else if (direction === "ArrowDown") {
+          newPosition.y += 20;
+        } else if (direction === "ArrowLeft") {
+          newPosition.x -= 20;
+        } else if (direction === "ArrowRight") {
+          newPosition.x += 20;
+        }
+
+        if (
+          newPosition.x === foodPosition.x &&
+          newPosition.y === foodPosition.y
+        ) {
+          let newFoodPosition = generateNewFoodPosition();
+          while (checkNewFoodOverlap(newFoodPosition)) {
+            newFoodPosition = generateNewFoodPosition();
+          }
+          setFoodPosition(newFoodPosition);
+          setScore(score + 1);
+          if (score >= highscore) {
+            setHighscore(score + 1);
+          }
+
+          setChildren((prevChildren) => [
+            ...prevChildren,
+            { x: prevPosition.x, y: prevPosition.y },
+          ]);
+        }
+
+        if (checkGameLost(newPosition, children)) {
+          setGameOn(false);
+          return prevPosition;
+        }
+        moveChildren();
+        setPrevDirection(direction);
+        return newPosition;
+      });
+    }
+
+    const moveInterval = setInterval(movePlayer, 100);
```

Comment:
Of course, I had to try out the game myself! I noticed that the snake moves very fast, and at some point, I got frustrated because I could never get more than one child. My suggestion is to slow it down a bit for a better user experience.

```suggestion
    const moveInterval = setInterval(movePlayer, 200);
```